client/lib/plans/index.js

```diff
@@ -214,11 +198,45 @@ export function getYearlyPlanByMonthly( planSlug ) {
  * @return {Boolean}           Whether the plan "types" match regardless of interval
  */
 export function planLevelsMatch( planSlugA, planSlugB ) {
-	return (
-		planSlugA === planSlugB ||
-		getMonthlyPlanByYearly( planSlugA ) === planSlugB ||
-		planSlugA === getMonthlyPlanByYearly( planSlugB )
-	);
+	const planA = getPlan( planSlugA );
+	const planB = getPlan( planSlugB );
+	return planA && planB && planA.type === planB.type && planA.group === planB.group;
+}
+
+export function isBusinessPlan( planSlug ) {
+	return getPlan( planSlug ).type === TYPE_BUSINESS;
+}
+
+export function isPremiumPlan( planSlug ) {
+	return getPlan( planSlug ).type === TYPE_PREMIUM;
+}
+
+export function isPersonalPlan( planSlug ) {
+	return getPlan( planSlug ).type === TYPE_PERSONAL;
+}
+
+export function isFreePlan( planSlug ) {
+	return getPlan( planSlug ).type === TYPE_FREE;
+}
+
+export function findSimilarPlan( planKey, query = {} ) {
+	const plan = getPlan( planKey );
+	const type = query.type || plan.type;
+	const group = query.group || plan.group;
+	const term = query.term || plan.term;
+
+	return getPlans().filter( p => planMatches( p, { type, group, term } ) )[ 0 ];
+}
+
+export function planMatches( planKey, query = {} ) {
+	const unknownKeys = difference( Object.keys( query ), [ 'type', 'group', 'term' ] );
+	if ( unknownKeys.length ) {
```

Comment:
I don't think we need to reject the extra keys here. Can we just ignore them? Or, please specify the error message if you think it is really necessary.